### ISSUE TITLE
Dependencies missing from pom

### ISSUE THREAD

**Issue by hvesalai** [haraldk/TwelveMonkeys#91]
Hi,

reading the documentation I came to understand that it would be enough for me to just include the following dependency to my project.

```xml
    <dependency>
        <groupId>com.twelvemonkeys.imageio</groupId>
        <artifactId>imageio-jpeg</artifactId>
        <version>3.0.1</version> <!-- Alternatively, build your own 3.1-something version -->
    </dependency>
```

However, it was not enough. `imageio-jpeg` does not include directly or indirectly the following required modules: `common-lang`, `common-io`, `common-image`

Looking at the poms the dependencies are only:
* imageio-jpeg
    * imageio-core
        * (no declared dependencies)
    * imageio-metadata
        * imageio-core
            * (no declared dependencies)

I.e. the mentioned required jars are not in the transitive closure of imageio-jpeg dependencies.

**Comment by haraldk** [comment]
Hi Heikki,

Thanks for reporting! 

The issue you are mentioning is probably caused by my still limited understanding of the mysterious ways Maven works... The dependencies you are mentioning are declared in the `imageio` pom, but seems this does not work then.

Would it be enough to add them to `imageio-core` (as everything imageio depends on this) and let them be transitive dependencies? 

PS: You should really be using 3.0.2, but it seems I forgot updating the version numbers last time I deployed a new version (fixed it right now, thanks ;-)).

Best regards,

-- 
Harald K

**Comment by hvesalai** [comment]
Ok, I see those parent declarations. Didn't notice them before. I'm actually not using `maven`, but `sbt` (a build toold for scala). So it might just be that `sbt` doesn't understand those parent projects. I'll debug more.

**Comment by hvesalai** [comment]
Ok, so I debugged more and `sbt` is working correctly and actually only the `common-lang` is missing. The reason is simply that if you look at https://github.com/haraldk/TwelveMonkeys/blob/master/imageio/pom.xml, you have defined `common-lang` only in the `test` scope. This means that it will not be available at runtime. 

The fix is equally simple, just make dependency on `common-lang` in the default scope as you have done for `common-io`. I.e. include this to the parent pom:

```xml
    <dependency>
       <groupId>com.twelvemonkeys.common</groupId>
       <artifactId>common-lang</artifactId>
       <version>${project.version}</version>
    </dependency>
```

**Comment by hvesalai** [comment]
Hmm... Not so fast, I'm also missing the `imageio-core`.

This is the list of stuff `sbt` resolves starting from the `imageio-jpeg` module:

```
com.twelvemonkeys.imageio#imageio-jpeg;3.0.2
com.twelvemonkeys.imageio#imageio;3.0.2
com.twelvemonkeys.imageio#imageio-metadata;3.0.2
com.twelvemonkeys.imageio#imageio;3.0.2
com.twelvemonkeys.common#common-io;3.0.2
com.twelvemonkeys.common#common-image;3.0.2
```

My bet is that the `imageio-core` is missing because the `imageio/pom.xml` contains two definitions for it. There's the one with classifier `testing` and the other without any classifier. Now when you refer to the module in `imageio-jpeg/pom.xml`, the reference is ambiguous. You should probably remove the `testing` version from the `imageio/pom.xml`.

**Comment by haraldk** [comment]
Hi again, 

Thanks for debugging this further.

Actually, there's nothing wrong or ambiguous with the seemingly duplicate dependency. The one with the `tests` classifier is really a different artifact altogether and needs to be included (some tests extend tests in other modules, so I depend on the *-tests.jar) in `test` scope only.

Also, both `common-io` and `common-image` depend on `common-lang` of the same version, so I don't see how that can not be included (as transitive dependency)?

In any case, I've have added `commons-lang` as a dependency from `imageio` and added explicit dependencies on `common-lang`, `common-io` and `common-image` from `imageio-core` to see if that helps. I'll check it in later.


-- 
Harald K

**Comment by haraldk** [comment]
Please do a pull now, and see if that helps. 

Checked in as 025021442f17c0901debe3ed9102782bca83cda5. 

These changes will be released as 3.1 at some point.

-- 
Harald K

**Comment by haraldk** [comment]
Closing for now, feel free to reopen, if there's still problems.

-- 
Harald K
